### PR TITLE
Mutect2 2.0

### PR DESCRIPTION
**Pull request by lkhilton**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [x] I used the cookiecutter template and updated the placeholder rules.

- [x] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules). 

  - [x] All references to the `rules` object (_e.g._ for input files) are wrapped with `str()`.

- [x] Every rule in the module is either listed under `localrules` or has the `threads` and `resources` directives.

- [x] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [x] I updated the final target rule (`*_all`) to include every output rule.

- [x] I explained important module design decisions in `CHANGELOG.md`.

- [x] I tested the module on real data for all supported `seq_type` values.

- [x] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [x] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository).

- [x] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [x] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### If applicable

- [x] I added more granular output subdirectories.

- [x] I added rules to the `reference_files` workflow to generate any new reference files.

- [x] I added subdirectories with large intermediate files to the list of `scratch_subdirectories` in the `default.yaml` configuration file.

- [x] I updated the list of available wildcards for the input files in the `default.yaml` configuration file.

## Checklist for Updated Module

To be completed.

### CODE REVIEW

**Comment by lkhilton** [comment]
*IMPORTANT* Before merging this PR I will be moving these files to a v2.0 of the mutect2 module and v 2.5 of the reference file workflow, so please don't merge after reviewing/approving. 

This version uses the latest mutect2 LearnReadOrientModel functionality for artifact filtering. I've also updated the reference file workflow to download from the Broad Google Cloud bucket because they will be deprecating the FTP download site.

**Comment by rdmorin** [comment]
@ppararaj are you reviewing this for Laura, Please? We need this module for GAMBL and BLGSP

**Review by ppararaj — APPROVED** [review]
Looks good

**Comment by lkhilton** [comment]
Ready for final approval!

File: modules/vcf2maf/1.1/vcf2maf.smk
Comment (rdmorin) [inline modules/vcf2maf/1.1/vcf2maf.smk:80]: Should we add this same logic for the intermediate vep file? If it is not deleted by the rule then it actually prevents VEP from re-running. VCF2maf just assumes that it's been run and uses the old vep.vcf as input.

File: workflows/reference_files/2.4/reference_files_header.smk
Comment (rdmorin) [inline workflows/reference_files/2.4/reference_files_header.smk:296]: I'm not telling you to not do this but some of us cannot understand what these awk commands do, so I'm simply taking your word for it that this works. Since I do stuff in Perl here and there I know I'm not in a good position to complain.

**Review by rdmorin — APPROVED** [review]
Thanks for those last few tweaks.

**Comment by rdmorin** [comment]
@lkhilton I leave it to you to merge this

**Pull request merged**
